src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -191,14 +193,20 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
     var si = 0
     var ei = 0
     var depth = 0
+
+    def checkEmptyTag(currentLetter: Int, position: Int): Boolean = {
```

Comment:
Will take another look for this logic.

Reply:
?

Reply:
Logic is usualy you try find this end `</rootTag>`, i add check and for `</>`.
Each time we check both of this situation.
If you any suggestions for this logic i think it's better do on refactor in next version.

Reply:
Yea, got it but I simply meant just double checking :).